### PR TITLE
sync `epic/ia` with `trunk`

### PR DESCRIPTION
**Pull request by jaredrethman**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Sync `epic/ia` with `trunk`. Changes (introduced in #3582) to Wizards needed to be migrated to new locations, some notable changes (13d6d55f6a3949ed17118a98369525cfc12ab442):
* Endpoint & callback for route `DELETE:/wizard/newspack-reader-revenue-wizard/reader-activation/emails/(?P<id>\d+)`  moved from `includes/wizards/class-reader-revenue-wizard.php` to `includes/wizards/audience/class-audience-donations.php`.
* Endpoint & callback for route `DELETE:/wizard/newspack-engagement-wizard/reader-activation/emails/(?P<id>\d+)` moved from `includes/wizards/class-engagement-wizard.php` to `includes/wizards/audience/class-audience-wizard.php`.
* Order notes checkbox added to `<BillingFields />` in `src/wizards/audience/components/billing-fields/index.tsx`
* "Checkout Configuration" group of components moved to Audience > Setup > Checkout & Payment.
* Audience > Setup tabs got wrapped in `WizardsSection`.
* Email reseting added to Transactional Emails & Newspack Settings > Emails.
* Recaptcha payload changes to accommodate `newspack_recaptcha_credentials ` option.

### How to test the changes in this Pull Request:

1. Checkout and build assets `git checkout feat/epic-ia-trunk-rasacc && npm run build`
2. Test email reseting:
   * Audience >  Setup > Transactional Emails
      * Navigate to _/wp-admin/admin.php?page=newspack-audience#/transactional-emails_
      * Pick an email and take note of the post id associated with it. This can be done by hovering over the "Edit" link.
      * Click "Reset", once request has completed, confirm the email has been reset (I did this by observing that the post id on the "Edit" link had changed).
Screenshot 2024-12-04 at 14 49 31
   * Newspack > Settings > Emails:
      * Navigate to _/wp-admin/admin.php?page=newspack-settings#/emails_
      * Observe "Cancellation" & "Welcome" emails have been added.
      * Follow the above repeatable steps starting at "Pick an email".
3. Navigate to _/wp-admin/admin.php?page=newspack-audience#/payment_
   * Scroll down to "Billing Fields".
       * Confirm newly added "Order notes" checkbox exists.
       * Perform editing & saving of checkbox and confirm changes persist between updates.
   * Scroll down to "Checkout Configuration"
       * Perform editing & saving of toggle and textareas and confirm changes persist between updates.
4. Navigate to _/wp-admin/admin.php?page=newspack-settings&scrollTo=newspack-settings-recaptcha#/_
   * Perform updates to recaptcha settings and confirm data persists between saves.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Great job parsing through all the conflicts @jaredrethman!

I tested the RAS-ACC flows (both as admin and as reader) and everything is working for the most part. I just ran into one issue with recaptcha. As admin, changing the recaptcha version causes the following browser error:

```
12:59:32.121 Uncaught TypeError: g is undefined
    pe https://site.test/wp-content/plugins/newspack-plugin/dist/newspack-wizards.js:26
    renderWithHooks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:15496
    updateFunctionComponent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:19627
    beginWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:21650
    beginWork$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:27475
    performUnitOfWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26606
    workLoopSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26515
    renderRootSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26483
    performSyncWorkOnRoot https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26134
    flushSyncCallbacks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12052
    flushSyncCallbacksOnlyInLegacyMode https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12031
    batchedUpdates$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26197
    batchedUpdates https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:3978
    dispatchEventForPluginEventSystem https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:9297
    dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6475
    dispatchEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6467
    dispatchDiscreteEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6440
newspack-wizards.js:26:5995
```
---

I also ran into a few issues that also exist on the `epic/ia` branch so arent blockers.

When I edit any email template, then go back via the newspack logo, I am directed to `/wp-admin/admin.php?page=newspack` and am shown the following error:

Screenshot 2024-12-05 at 12 32 55

There is no loading state when saving billing fields:

https://github.com/user-attachments/assets/b1452a0b-74f7-474b-a4e1-c4afa1cc2f27

File: src/wizards/audience/views/setup/payment.js [hunk 1]
```diff
@@ -2,28 +2,118 @@
...
 import NRHSettings from '../../components/nrh-settings';
 import BillingFields from '../../components/billing-fields';
 
-export default withWizardScreen( function () {
+export default withWizardScreen( function ( {
+	config,
+	getSharedProps,
+	updateConfig,
+	saveConfig,
+} ) {
```
Comment (miguelpeixe) [inline src/wizards/audience/views/setup/payment.js]: Suggestion: do you think it'd be too big of a refactor to move this to use the wizard store, as it's being used for 
`newspack-audience/payment`?

It allows for the component to not have to pass down all these props and the "Checkout Configuration" wizard section can become its own standalone component, like `<PaymentGateways />` and the others.

Using the store doesn't seem too different from what it's doing [here](https://github.com/Automattic/newspack-plugin/blob/13d6d55f6a3949ed17118a98369525cfc12ab442/src/wizards/audience/views/setup/index.js#L33-L66):

```js
const data = Wizard.useWizardData( 'newspack-audience/reader-activation' );

updateWizardSettings( {
	slug: 'newspack-audience/reader-activation'
} );

saveWizardSettings( {
	slug: 'newspack-audience/reader-activation'
} );
```

**Comment by jaredrethman** [comment]
@chickenn00dle thank you for the review and feedback, some responses below.

> I tested the RAS-ACC flows (both as admin and as reader) and everything is working for the most part. I just ran into one issue with recaptcha. As admin, changing the recaptcha version causes the following browser error:
> ```
> 12:59:32.121 Uncaught TypeError: g is undefined
>     pe https://site.test/wp-content/plugins/newspack-plugin/dist/newspack-wizards.js:26
>     renderWithHooks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:15496
>     updateFunctionComponent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:19627
>     beginWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:21650
>     beginWork$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:27475
>     performUnitOfWork https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26606
>     workLoopSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26515
>     renderRootSync https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26483
>     performSyncWorkOnRoot https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26134
>     flushSyncCallbacks https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12052
>     flushSyncCallbacksOnlyInLegacyMode https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:12031
>     batchedUpdates$1 https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:26197
>     batchedUpdates https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:3978
>     dispatchEventForPluginEventSystem https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:9297
>     dispatchEventWithEnableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6475
>     dispatchEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6467
>     dispatchDiscreteEvent https://site.test/wp-includes/js/dist/vendor/react-dom.js?ver=18.3.1:6440
> newspack-wizards.js:26:5995
> ```

I wasn't able to reproduce this, see below. Maybe `npm install --legacy-peer-deps && npm run build`?

https://github.com/user-attachments/assets/505ee94e-b0c9-4881-b546-0fbe6e8f9c31

> I also ran into a few issues that also exist on the `epic/ia` branch so arent blockers.
> 
> When I edit any email template, then go back via the newspack logo, I am directed to `/wp-admin/admin.php?page=newspack` and am shown the following error:
> 
> Screenshot 2024-12-05 at 12 32 55

Can you share some more details here please. I can't seem to find a "newspack logo" when editing an email template

Screenshot 2024-12-05 at 22 16 14

> There is no loading state when saving billing fields:
> 
>  Screen.Recording.2024-12-05.at.12.51.35.mov

This is a trade-off when using `Wizard.useWizardData`, loading states need to be handled on a per/component bases. I've added some visual feedback in ca290e1545d31becb749ec4159bc43008dd27057.

https://github.com/Automattic/newspack-plugin/blob/d2ce4ffc321bb87cd511fe882cadb50605656253/src/wizards/audience/components/billing-fields/index.tsx#L14-L17

https://github.com/user-attachments/assets/cc72cf9b-8ec1-4aa6-9ff8-ce368178c8f5

File: src/wizards/audience/views/setup/payment.js (hunk 1 above)
Comment (jaredrethman) [reply src/wizards/audience/views/setup/payment.js]: @miguelpeixe thank you for the review and suggestion. See refactor(s) in 58020abef2b88a5f5430a01173616ef629c3b852, c5f5e6110d7cd68a898e4eb682718ab5577d9897, 3108cde1e004fad558d00cc6216d52ee3382bb2f, 0f6f5cef993f83e1792fde4b84c827a289155a23

**Comment by chickenn00dle** [comment]
Thanks for the quick response @jaredrethman!

I'll have another look at building a bit later, but in the meantime:

> Can you share some more details here please. I can't seem to find a "newspack logo" when editing an email template

Sorry about that. I meant to say site logo. Its in the top left corner to return to the settings page:
Screenshot 2024-12-06 at 12 20 59

**Comment by jaredrethman** [comment]
> Thanks for the quick response @jaredrethman!
> 
> I'll have another look at building a bit later, but in the meantime:
> 
> > Can you share some more details here please. I can't seem to find a "newspack logo" when editing an email template
> 
> Sorry about that. I meant to say site logo. Its in the top left corner to return to the settings page: Screenshot 2024-12-06 at 12 20 59

Thanks for the screenshot. ff0ffd367da39257da7ec910a55c8a2db6d9f605 fixes the bug!

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Just tested this again and am still running into issues after reinstall with legacy deps and building. The error seems to be coming from this file: https://github.com/Automattic/newspack-plugin/blob/ff0ffd367da39257da7ec910a55c8a2db6d9f605/src/wizards/newspack/views/settings/connections/recaptcha.tsx#L276

Looks like `settingsToUpdate` initializes with only v3 settings for me (likely because I emptied my v2 keys a while back) and so switching from v3 to v2 causes `versionCredentials` to be undefined. Guarding against this for both `site_key` and `site_secret` fixes the error for me: `versionCredentials?.site_key`/`versionCredentials?.site_secret`.

Although I then see a new issue where the inputs are no longer updating.

**Comment by jaredrethman** [comment]
> Just tested this again and am still running into issues after reinstall with legacy deps and building. The error seems to be coming from this file:
> 
> https://github.com/Automattic/newspack-plugin/blob/ff0ffd367da39257da7ec910a55c8a2db6d9f605/src/wizards/newspack/views/settings/connections/recaptcha.tsx#L276
> 
> Looks like `settingsToUpdate` initializes with only v3 settings for me (likely because I emptied my v2 keys a while back) and so switching from v3 to v2 causes `versionCredentials` to be undefined. Guarding against this for both `site_key` and `site_secret` fixes the error for me: `versionCredentials?.site_key`/`versionCredentials?.site_secret`.
> 
> Although I then see a new issue where the inputs are no longer updating.

Fixed in 61c9e1525a4fcc5c6250fe6e78a3a4e0782e5e6b. Change is mainly to keep the UI code clean, and making the API more consistent supports this better.

**Review by chickenn00dle — APPROVED** [review]
Sorry for the delay on this! All good on my end 👍

File: src/wizards/audience/views/setup/payment.js (hunk 1 above)
Comment (miguelpeixe) [reply src/wizards/audience/views/setup/payment.js]: Thank you for the refactor, @jaredrethman! It works great.

**Review by miguelpeixe — APPROVED** [review]
Thank you for the revisions! It's working well and I couldn't find any regressions in other wizards 🎉

**Pull request merged**